Source/missiles.cpp

```diff
@@ -3685,6 +3685,35 @@ void MI_Fireball(int i)
 #ifdef HELLFIRE
 void missiles_4359A0(int i)
 {
+	int t, j, k, tx, ty, dp;
+
+	missile[i]._mirange--;
+	CheckMissileCol(i, 0, 0, 0, missile[i]._mix, missile[i]._miy, 0);
+	if (missile[i]._mirange <= 0) {
+		missile[i]._miDelFlag = TRUE;
+		for (j = 0; j < 2; j++) {
+			k = CrawlNum[j] + 2;
+			for (t = CrawlTable[CrawlNum[j]]; t > 0; t--, k += 2) {
+				tx = missile[i]._mix + CrawlTable[k - 1];
+				ty = missile[i]._miy + CrawlTable[k];
+				if (tx > 0 && tx < MAXDUNX && ty > 0 && ty < MAXDUNY) {
+					dp = dPiece[tx][ty];
+					if (!nSolidTable[dp] && !dMonster[tx][ty] && !dPlayer[tx][ty] && !dObject[tx][ty]) {
+						j = 6;
+						int mon = AddMonster(tx, ty, missile[i]._miVar1, 1, TRUE); // TODO: apply enum - haven't managed to figure out this 1

```

Comment:
This is the index in the `Monsters` array, it doesn't really make sense for it to have an enum.

```suggestion
						int mon = AddMonster(tx, ty, missile[i]._miVar1, 1, TRUE);
```

Reply:
so what does this do? summon a zombie?

Reply:
Monsters is populated at run time, there is no fixed data for it.

It spawns a monster that is of what ever type has been set as index 1 for that level, on that run https://github.com/diasurgical/devilution/blob/0704393bc3396f3d190da8f2c7ff879100058b4b/Source/monster.cpp#L380

`AddMonsterType` is what normal populates that array. For leoric its a random skeleton https://github.com/diasurgical/devilution/blob/0704393bc3396f3d190da8f2c7ff879100058b4b/Source/monster.cpp#L232
for level 16 it would be an advocate https://github.com/diasurgical/devilution/blob/0704393bc3396f3d190da8f2c7ff879100058b4b/Source/monster.cpp#L195

Reply:
still, what does this function do? xD I'm curious lol

Reply:
Looks like a spell that can create monsters, probably it's related to the Hork Demon https://youtu.be/rp8cOiTkQH0?t=129